src/Control/Distributed/Process/Tests/CH.hs

```diff
@@ -1229,17 +1229,19 @@ testExitLocal TestTransport{..} = do
   localNode <- newLocalNode testTransport initRemoteTable
   supervisedDone <- newEmptyMVar
   supervisorDone <- newEmptyMVar
+  l <- newEmptyMVar
 
   pid <- forkProcess localNode $ do
-    (liftIO $ threadDelay 100000)
-      `catchExit` \_from reason -> do
+    (liftIO (putMVar l ()) >> expect) `catchExit` \_from reason -> do
         -- TODO: should verify that 'from' has the right value
         True <- return $ reason == "TestExit"
         liftIO $ putMVar supervisedDone ()
 
   runProcess localNode $ do
+    liftIO $ takeMVar l
```

Comment:
> Wouldn't it be better to fix forkProcess and call it with exceptions masked?
> If not, then it would be nice adding your line response as a comment.

I think it should be run with inherited mask state (it's not done yet), and possibly
with adding `forkProcessWithUnmask`